### PR TITLE
Nitpretty: make the tool more customizable

### PR DESCRIPTION
**Pull request by Morriar**
Added some options to control behavior.

Interesting features:

* option --break-strings to enable literal string breaks
* option --inline-do to enable do inlining
* option --skip-empty to enable empty line skipping

### CODE REVIEW

File: tests/sav/nitpretty_args1.res [hunk 1]
```diff
@@ -12,4 +12,4 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# An empty module
+# An empty module
```
Comment (privat) [inline tests/sav/nitpretty_args1.res]: ajouter un \n sur la derniere ligne pour faire plaisir a git?

**Comment by Morriar** [comment]
Reroll avec tes commentaires:

1. j'ai retiré le commit qui enlevait le dernier `\n`
2. le dernier commit s'occupe de rajouter un `\n` uniquement si `--skip-empty` est activé

**Comment by privat** [comment]
+1

**Comment by Morriar** [comment]
Ping @xymus, @R4PaSs, @BlackMinou go mettre un +1.

File: src/nitpretty.nit [hunk 2]
```diff
@@ -32,6 +32,16 @@ redef class ToolContext
 	var opt_meld = new OptionBool("Show diff between source and output using meld",
 	   "--meld")
 
+	# Break too long string literals.
+	var opt_break_str = new OptionBool("Break too long string literals", "--break-strings")
+
+	# Force `do` on same line than method signature.
+	var opt_inline_do = new OptionBool("Force do keyword on the same line than the method signature",
```
Comment (R4PaSs) [inline src/nitpretty.nit]: :s/same line than/same line as

File: src/nitpretty.nit [hunk 3]
```diff
@@ -32,6 +32,16 @@ redef class ToolContext
 	var opt_meld = new OptionBool("Show diff between source and output using meld",
 	   "--meld")
 
+	# Break too long string literals.
+	var opt_break_str = new OptionBool("Break too long string literals", "--break-strings")
+
+	# Force `do` on same line than method signature.
```
Comment (R4PaSs) [inline src/nitpretty.nit]: :s/line than/line as

File: src/pretty.nit [hunk 4]
```diff
@@ -223,6 +223,14 @@ class PrettyPrinterVisitor
 		if current_length == 0 and last_line_is_blank then return
 		previous_length = current_length
 		current_length = 0
+		if skip_empty then wait_addn += 1
+	end
+
+	# Force the line to end even is not skip_empty.
+	fun forcen do
```
Comment (R4PaSs) [inline src/pretty.nit]: Forcen ? Strange name for a method, I don't understand its purpose either, event with the comment, care to explain ?

File: src/pretty.nit [hunk 5]
```diff
@@ -243,6 +251,15 @@ class PrettyPrinterVisitor
 			consume "."
 		end
 	end
+
+	# Do we break string literals that are too long?
+	var break_strings = false is public writable
+
+	# Do we force `do` on the same line than method signature?
```
Comment (R4PaSs) [inline src/pretty.nit]: :s/than method/as the method

**Comment by xymus** [comment]
I don't know exactly what the option `--skip-empty` does, but besides that +1.

**Comment by R4PaSs** [comment]
+1 aside from my comments

**Comment by Morriar** [comment]
Reroll:

* removed useless empty line at pretty.nit:838
* corrected typos as denoted by @R4PaSs 
* changed comment for `forcen`
* enhanced comment for `--skip-empty`

**Pull request merged**